packages/website/src/server/index.ts

```diff
@@ -152,12 +168,59 @@ export async function startWebsite(options: IStartWebisteOptions) {
     hotClient
   }));
 
-  const server = koa.listen({
-    host: options.host,
-    port: options.port
-  });
+  const server = koa.listen({host, port});
 
   server.on('listening', () => {
-    process.stdout.write(`Listening on ${getServerUrl(server)}\n`);
+    const serverUrl = getServerUrl(server);
+    process.stdout.write(`Running on ${chalk.blue(serverUrl)}\n`);
   });
 }
+
+export async function buildWebsite(
+  {projectOptions, outputPath}: IBuildWebsiteOptions
+) {
+  try {
+    // Compile the metadata first, because if there are type or syntactic errors
+    // in the project it's easier to let Webpack catch and report them instead
+    // of implementing friendly error reporting in the schema extractor.
+
+    const metadataCompiler = webpack(getMetadataWebpackConfig({
+      projectOptions,
+      production: true,
+      outputPath
+    }));
+
+    await new Promise((resolve, reject) => {
```

Comment:
this could be a good case for `promisify` https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original